ui/app/components/totp/key-create.hbs

```diff
@@ -7,11 +7,15 @@
   <div class="box is-sideless is-fullwidth is-marginless">
     <MessageError @model={{@model}} />
     <NamespaceReminder @mode={{@mode}} @noun="TOTP key" />
-    <FormFieldGroups
-      @model={{@model}}
-      @mode={{@mode}}
-      @groupName={{if @model.generate "generatedFormFieldGroups" "nonGeneratedFormFieldGroups"}}
-    />
+
+    {{#each (if @model.generate @generatedFields @nonGeneratedFields) as |field|}}
```

Comment:
Nice! This is a lot cleaner and easier to follow. If you follow the [suggestion above](https://github.com/hashicorp/vault/pull/29805/files/a6613f17707b24121eae76b06ba44143c1c2ee1e..df6bf0427da7a103a4d88811906d756cc6744e7e#r1985605544), we can compute this in the component class instead and really clean this up to just be. 

```suggestion
    {{#each this.keyFormFields as |field|}}
```